gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -287,22 +287,75 @@ function plymeta:CanCarryWeapon(wep)
 	if not wep or not wep.Kind then
 		return false
 	end
-
+	
+	--appeareantly TTT can't handle two times the same weapon
+	for k, v in pairs(self:GetWeapons()) do
+		if IsValid(wep) and IsValid(v) and v:GetClass() == wep:GetClass() then
+			return false
+		elseif istable(wep) and IsValid(v) and v:GetClass() == wep.ClassName then
```

Comment:
This elseif should never get called...

Reply:
Actually it is called when a player is getting a new weapon (e.g. via Shop/Console)